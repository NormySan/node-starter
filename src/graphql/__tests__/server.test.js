import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';

import { createTestServer } from './createTestServer';

it('fetches an array of books', async () => {
  const GET_BOOKS = gql(`
    query {
      books {
        id
        title
        author
      }
    }
  `);

  // Create a test client with a mock server.
  const { query } = createTestClient(createTestServer());

  // Run the query against the test server and snapshot the output.
  const response = await query({ query: GET_BOOKS });
  expect(response).toMatchSnapshot();
});

it('fetches a single book', async () => {
  const GET_BOOK = gql(`
    query {
      book(id: "1") {
        id
        title
        author
      }
    }
  `);

  // Create a test client with a mock server.
  const { query } = createTestClient(createTestServer());

  // Run the query against the test server and snapshot the output.
  const response = await query({ query: GET_BOOK });
  expect(response).toMatchSnapshot();
});

it('returns null when no book is found', async () => {
  const GET_BOOK_NOT_FOUND = gql(`
    query {
      book(id: "10") {
        id
        title
        author
      }
    }
  `);

  // Create a test client with a mock server.
  const { query } = createTestClient(createTestServer());

  // Run the query against the test server and snapshot the output.
  const response = await query({ query: GET_BOOK_NOT_FOUND });
  expect(response).toMatchSnapshot();
});
