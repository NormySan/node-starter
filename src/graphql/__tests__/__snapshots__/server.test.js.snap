// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches a single book 1`] = `
Object {
  "data": Object {
    "book": Object {
      "author": "J.K. Rowling",
      "id": "1",
      "title": "Harry Potter and the Chamber of Secrets",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches an array of books 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "author": "J.K. Rowling",
        "id": "1",
        "title": "Harry Potter and the Chamber of Secrets",
      },
      Object {
        "author": "Michael Crichton",
        "id": "2",
        "title": "Jurassic Park",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`returns null when no book is found 1`] = `
Object {
  "data": Object {
    "book": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
